cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/AggregatedMetricsCollectionService.java

```diff
@@ -75,9 +78,11 @@ public AggregatedMetricsEmitter load(EmitterKey key) throws Exception {
    * This method is guaranteed not to get concurrent calls.
    *
    * @param metrics collection of {@link co.cask.cdap.api.metrics.MetricValue} to publish.
+   * @param metaMetricsComputer computes metrics about metrics
    * @throws Exception if there is error raised during publish.
    */
-  protected abstract void publish(Iterator<MetricValue> metrics) throws Exception;
+  protected abstract void publish(Iterator<MetricValue> metrics,
+                                  MetaMetricsComputer metaMetricsComputer) throws Exception;
```

Comment:
Why need to change this? Isn't it better to have the Iterator do it internally while iterating, rather than sub-class need to calls it explicitly?